securedrop/source.py

```diff
@@ -347,6 +347,20 @@ def delete():
     flash("Reply deleted", "notification")
     return redirect(url_for('lookup'))
 
+@app.route('/delete-all', methods=('POST',))
+@login_required
+def batch_delete():
+    replies = Reply.query.filter(Reply.source_id == g.source.id).all()
```

Comment:
Since `Reply` defines a backref to `Source`, you can make this a little simpler: `replies = g.source.replies` should be equivalent.